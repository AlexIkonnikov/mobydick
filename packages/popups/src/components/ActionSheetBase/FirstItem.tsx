import React, {FC} from 'react';
import {useStyles} from '@npm/mobydick-styles';

import stylesCreate from './stylesCreate';
import Item, {IPropsItem} from './Item';
import Label, {IPropsLabel} from './Label';

const FirstItem: FC<IPropsItem | IPropsLabel> = props => {
  const [styles] = useStyles(stylesCreate);
  const {label, ...otherProps} = props;
  return label ? (
    <Label style={styles.firstItem} label={label} {...otherProps} />
  ) : (
    <Item style={styles.firstItem} {...otherProps} />
  );
};

export default FirstItem;
