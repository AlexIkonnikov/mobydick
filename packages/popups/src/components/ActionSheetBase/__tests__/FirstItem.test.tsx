import {render} from '@testing-library/react-native';
import React from 'react';

import FirstItem from '../FirstItem';

describe('@npm/mobydick-popups/ActionSheetBase', () => {
  it('should renders correctly FirstItem', () => {
    const {toJSON} = render(<FirstItem />);

    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly FirstItem label', () => {
    const {toJSON} = render(
      <FirstItem label={'label'} labelFont={'Regular-White-S'} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
