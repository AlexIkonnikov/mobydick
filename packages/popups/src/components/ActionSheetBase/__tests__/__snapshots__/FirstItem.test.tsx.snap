// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-popups/ActionSheetBase should renders correctly FirstItem 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EEEFF1",
      "borderTopLeftRadius": 18,
      "borderTopRightRadius": 18,
      "justifyContent": "center",
      "marginHorizontal": 12,
      "minHeight": 75,
      "opacity": 1,
      "paddingHorizontal": 30,
    }
  }
>
  <Text
    allowFontScaling={true}
    style={
      Array [
        Object {
          "color": "#20242D",
          "fontFamily": "Inter-Regular",
          "fontSize": 27,
          "lineHeight": 33,
        },
        Object {
          "paddingVertical": 22.5,
        },
      ]
    }
  />
</View>
`;

exports[`@npm/mobydick-popups/ActionSheetBase should renders correctly FirstItem label 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#EEEFF1",
        "justifyContent": "center",
        "marginHorizontal": 12,
        "minHeight": 75,
        "paddingHorizontal": 30,
      },
      Object {
        "borderTopLeftRadius": 18,
        "borderTopRightRadius": 18,
      },
    ]
  }
>
  <Text
    allowFontScaling={true}
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontFamily": "Inter-Regular",
          "fontSize": 22.5,
          "lineHeight": 28.5,
        },
        Object {
          "paddingVertical": 22.5,
          "textAlign": "center",
        },
      ]
    }
  >
    label
  </Text>
</View>
`;
